### PR TITLE
fix(headless)!: clean up deprecated feedback methods

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5585](https://coveord.atlassian.net/browse/SFINT-5585)

- Updated `sendFeedback`  method in the generated answer controllers to accept only GeneratedAnswerFeedbackV2, the latest feedback schema we created to represent a feedback.
- Updated the  analytic actions logGeneratedAnswerFeedback by remove the logic that checks  between v1 and v2 and use only GeneratedAnswerFeedbackV2 type.
- Removed logGeneratedAnswerDetailedFeedback analytics action, as it is no longer used.
- Renamed the type `GeneratedAnswerFeedbackV2` to `GeneratedAnswerFeedback`

[SFINT-5585]: https://coveord.atlassian.net/browse/SFINT-5585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/controllers/core/generated-answer/headless-searchapi-generated-answer.test.ts [hunk 1]
```diff
@@ -124,26 +124,34 @@ describe('searchapi-generated-answer', () => {
...
-  it('dispatches a send feedback action', () => {
+  // it('dispatches a send feedback action', () => {
+  //   const generatedAnswer = createGeneratedAnswer();
+  //   const feedback: GeneratedAnswerFeedback = 'harmful';
+  //   generatedAnswer.sendFeedback(feedback);
+  //   expect(
+  //     generatedAnswerAnalyticsClient.logGeneratedAnswerFeedback
+  //   ).toHaveBeenCalledWith(feedback);
+  //   expect(sendGeneratedAnswerFeedback).toHaveBeenCalledTimes(1);
+  // });
```
Comment (erocheleau) [inline packages/headless/src/controllers/core/generated-answer/headless-searchapi-generated-answer.test.ts:135]: TODO?

**Review by erocheleau — COMMENTED** [review]
looks good, just commented code :)

File: packages/headless/src/controllers/core/generated-answer/headless-searchapi-generated-answer.test.ts (hunk 1 above)
Comment (mmitiche) [reply packages/headless/src/controllers/core/generated-answer/headless-searchapi-generated-answer.test.ts]: this is no longer needed, forgot to remove 👍

**Review by SimonMilord — APPROVED** [review]
LGTM!

**Review by louis-bompart — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**